### PR TITLE
Trigger modifications and general cleanup

### LINKED ISSUE #97: Create new swagger doc with updated triggers and server size definitions
Clarify trigger definitions:

- Eliminate triggers endpoint
- Move services to project/servers/triggers
- Rename resources to server-size
- Modify docs so super users in on-premise version only can modify server size

3blades/3blades#118

### PR DESCRIPTION
**Pull request by jgwerner**
Closes #97 #100 #101

### CODE REVIEW

File: tbs_swagger.yml [hunk 1]
```diff
@@ -3474,65 +3211,176 @@ paths:
...
+          type: string
+        - name: id
+          required: true
+          in: path
+          description: 'Trigger identifier expressed as UUID.'
+          type: string
+      summary: "Delete a server trigger"
+      tags:
+        - service
+  '/v1/{namespace}/projects/{project_id}/servers/{server_id}/trigger/{id}/call':
```
Comment (johngriebel) [inline tbs_swagger.yml]: I'm not sure this endpoint should be exposed; I don't think triggers should be called manually. @jkp85 ?

**Comment by johngriebel** [comment]
@jgwerner looking again, I don't see anything about project triggers. Did we want to expose those now, or wait until later? I can't remember.

File: tbs_swagger.yml (hunk 1 above)
Comment (jkp85) [inline tbs_swagger.yml]: It depends what the goal is. If it should be a webhook then it can be exposed.

**Comment by raygesualdo** [comment]
Like John said, the diff is a little hard to grok. @jgwerner can you help me understand where the definitions are for the trigger payloads to/from the server? If I remember correctly from our meeting, those payloads were to look something like the following: 

```yaml
cause: enum based on trigger context e.g. START|TERMINATE for servers
effect: enum of allowed effects; start with raw webhook only
effectOptions: key/value pairs of effect-specific configuration
```

Did I misunderstand? Or am I missing it in the swagger file?

File: tbs_swagger.yml (hunk 1 above)
Comment (jgwerner) [inline tbs_swagger.yml]: @johngriebel This is related to exposing a trigger so that payloads originating from other systems can activate the trigger.

**Comment by jgwerner** [comment]
@johngriebel @jkp85 @raygesualdo I added this version to SwaggerHub so you can see the modified endpoints as API docs: https://swaggerhub.com/apis/3Blades/3blades_api/2.0

**Comment by jkp85** [comment]
@jgwerner Ok, How about user stories?

**Comment by jgwerner** [comment]
@jkp85 by user stories to you mean github issue?

**Comment by jkp85** [comment]
@jgwerner More I think about how user will use those triggers. Ex: user would want start a server after external system do its job (authenticated webhook).

**Comment by jgwerner** [comment]
@jkp85 use cases to issue #97

**Comment by jkp85** [comment]
@jgwerner Do we need a call counter for trigger? Like how many times particular trigger was executed?

**Comment by jgwerner** [comment]
@jkp85 @johngriebel I don't have any strong opinions about the counter. As long as we track actions I believe we can track what called the action, such as a particular trigger. So that should be enough, what do you think?

**Comment by jgwerner** [comment]
@johngriebel @raygesualdo @roachmd just pushed all changes. Please approve when possible to move forward with merge.

**Pull request merged**